find-external/Ipopt/FindIpopt.cmake

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 CNRS INRIA
+#
+# Author: Wilson Jallet
+#
+# Adapted from:
+# https://github.com/casadi/casadi/blob/main/cmake/FindIPOPT.cmake, LGPL 3.0
+# License
+#
+# Uses the modern PkgConfig CMake module helpers to find an installed version of
+# Ipopt, for which a CMake shared imported library target is created with the
+# required includes and compile options in its link interface.
+#
+find_package(Ipopt CONFIG QUIET)
+
+if(Ipopt_FOUND)
+  message(DEBUG "Found Ipopt (using IpoptConfig.cmake or ipopt-config.cmake)")
+else()
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(Ipopt REQUIRED ipopt)
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(
+    Ipopt
+    FAIL_MESSAGE DEFAULT_MSG
+    REQUIRED_VARS Ipopt_INCLUDE_DIRS Ipopt_LIBRARIES
+    VERSION_VAR Ipopt_VERSION)
+
+  message(STATUS "  Ipopt library dirs: ${Ipopt_LIBRARY_DIRS}")
+  message(STATUS "  Ipopt include dirs: ${Ipopt_INCLUDE_DIRS}")
+  add_library(ipopt SHARED IMPORTED)
+  # On Windows, ipopt library is named ipopt-3
+  find_library(
+    ipopt_lib_path
+    NAMES ipopt ipopt-3
+    PATHS ${Ipopt_LIBRARY_DIRS} REQUIRED)
+  message(STATUS "  Ipopt library found at ${ipopt_lib_path}")
+  set_target_properties(
+    ipopt
+    PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${Ipopt_INCLUDE_DIRS}"
+               INTERFACE_COMPILE_OPTIONS "${Ipopt_CFLAGS_OTHER}"
+               IMPORTED_CONFIGURATIONS "RELEASE")
+
+  if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    target_compile_definitions(ipopt INTERFACE HAVE_CSTDDEF)
+  endif()
```

Comment:
I don't know if this workaround is still needed : https://github.com/ghorn/debian-coinor-ipopt/issues/2

I test without it and it's working on conda-forge Ipopt.
Maybe it's mandatory for older Ipopt.

Reply:
Okay, perhaps we could remove it and ask for a given minimum version if Ipopt?